adapters/epics.py

```diff
@@ -25,75 +25,206 @@
+        This method forwards property read access on instance
+        to the member of instance that was selected in __init__.
+
+        :param instance: Instance of type.
+        :param type: Type.
+        :return: Attribute value of member property.
+        """
+        return getattr(getattr(instance, self._target_member), self._prop)
+
+    def __set__(self, instance, value):
+        """
+        This method forwards property write access on instance
+        to the member of instance that was selected in __init__.
+
+        :param instance: Instance of type.
+        :param value: Type.
+        """
+        setattr(getattr(instance, self._target_member), self._prop, value)
+
+
 class EpicsAdapter(Adapter):
-    def __init__(self, target, bindings, arguments):
-        super(EpicsAdapter, self).__init__(target, bindings, arguments)
+    """
+    Inheriting from this class provides an EPICS-interface to a device, powered by
+    the pcaspy-module. In the simplest case all that is required is to inherit
+    from this class and override the `pvs`-member. It should be a dictionary
+    that contains PV-names (without prefix) as keys and instances of PV as
+    values.
+
+    For a simple device with two properties, speed and position, the first of which
+    should be read-only, it's enough to define the following:
+
+        class SimpleDeviceEpicsAdapter(EpicsAdapter):
+            pvs = {
+                'VELO': PV('speed', read_only=True),
+                'POS': PV('position', lolo=0, hihi=100)
+            }
+
+    For more complex behavior, the adapter could contain properties that do not
+    exist in the device itself. If the device should also have a PV called STOP
+    that "stops the device", the adapter could look like this:
+
+        class SimpleDeviceEpicsAdapter(EpicsAdapter):
+            pvs = {
+                'VELO': PV('speed', read_only=True),
+                'POS': PV('position', lolo=0, hihi=100),
+                'STOP': PV('stop', type='int'),
+            }
+
+            @property
+            def stop(self):
+                return 0
+
+            @stop.setter
+            def stop(self, value):
+                if value == 1:
+                    self._device.halt()
+
+    Even though the device does _not_ have a property called 'stop' (but a method called 'halt'),
+    issuing the command
+
+        caput STOP 1
+
+    will achieve the desired behavior, because EpicsAdapter merges the properties
+    of the device into SimpleDeviceEpicsAdapter itself, so that it is does not
+    matter whether the specified property in PV exists in the device or the adapter.
+
+    The intention of this design is to keep device classes small and free of
+    protocol specific stuff, such as in the case above where stopping a device
+    via EPICS might involve writing a value to a PV, whereas other protocols may
+    offer an RPC-way of achieving the same thing.
+    """
+    protocol = 'epics'
+    pvs = None
```

Comment:
Yes, we discussed this, @MikeHart85 looked at the consequences of this with a minimal example, this is exactly the behavior that we want. Also, we'll never have two devices that use the same `EpicsAdapter`.